src/main/Main.ts

```diff
@@ -1,12 +1,52 @@
-import {app, ipcMain, nativeImage, BrowserWindow, screen} from "electron";
+import {app, nativeImage, BrowserWindow, screen, globalShortcut, Menu} from "electron";
 import {readFileSync} from "fs";
 import {windowBoundsFilePath} from "../utils/Common";
+import {buildMenuTemplate} from "./Menu";
+
+/**
+ * Contains references to all currently open windows.
+ *
+ * @type {Array}
+ */
+let windows: Electron.BrowserWindow[] = [];
```

Comment:
Okay I believe this will do it:

```TypeScript
//
// ─── ZIPPER CLASS ───────────────────────────────────────────────────────────────
//

    /** A _Set_ that has one __active__ element */
    class Zipper extends Set {

        //
        // ─── EXTRA STORAGE ───────────────────────────────────────────────
        //

            private activeElement: any;

        //
        // ─── CONSTRUCTOR ─────────────────────────────────────────────────
        //

            constructor ( ) {
                super( );
                this.activeElement = null;
            }

        //
        // ─── REMOVE ELEMENT ──────────────────────────────────────────────
        //

            public delete ( element ) {
                // if the active index is the index we want
                // to delete we have to change it.
                if ( this.activeElement === element ) {
                    // finding the active index
                    let activeIndex = 0;
                    for ( let index in this.values ) {
                        if ( this.values[ index ] === element ) {
                            this.activeElement = index;
                            break;
                        }
                    }

                    // checking what to do:
                    if ( activeIndex > 0 ) {
                        this.activeElement = this.values[ activeIndex - 1 ];
                    }

                    else if ( activeIndex < this.values.length ) {
                        this.activeElement = this.values[ activeIndex + 1 ];
                    }

                    else {
                        this.activeElement = null;
                    }
                }

                // super
                this.delete( element );
            }

        //
        // ─── SET ACTIVE INDEX ────────────────────────────────────────────
        //

            public activate ( element ) {
                if ( this.has( element ) ) {
                    this.activeElement = element;
                    return true;
                } else {
                    return false
                }
            }

        //
        // ─── GET FOCUSED ELEMENT ─────────────────────────────────────────
        //

            public getActiveElement ( ) {
                return this.activeElement;
            }

        // ─────────────────────────────────────────────────────────────────

    }

// ────────────────────────────────────────────────────────────────────────────────

```